http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -1194,6 +1209,94 @@ class GKEInterpreter[F[_]](
     } yield ()
   }
 
+  private[util] def installRStudioApp(
+    helmAuthContext: AuthContext,
+    appName: AppName,
+    release: Release,
+    cluster: KubernetesCluster,
+    nodepoolName: NodepoolName,
+    namespaceName: NamespaceName,
+    disk: PersistentDisk,
+    ksaName: ServiceAccountName,
+    gsa: WorkbenchEmail,
+    userEmail: WorkbenchEmail
+  )(implicit ev: Ask[F, AppContext]): F[Unit] = {
+    val chart = config.RStudioAppConfig.chart
+
+    for {
+      ctx <- ev.ask
+
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Installing helm chart for RStudio app ${appName.value} in cluster ${cluster.getClusterId.toString}"
+      )
+
+      googleProject <- F.fromOption(
+        LeoLenses.cloudContextToGoogleProject.get(cluster.cloudContext),
+        new RuntimeException("trying to create an azure runtime in GKEInterpreter. This should never happen")
+      )
+
+      // Create the staging bucket to be used by Welder
+      stagingBucketName = generateUniqueBucketName("leostaging-" + appName.value)
```

Comment:
Does welder actually use the staging bucket for anything? I know it talks to the workspace bucket but I don't recall a dependency on the extra staging bucket.

Just wondering if this migrated from the GCE installation -- we could avoid a bucket creating if we can be sure welder doesn't use it.

Reply:
I might have misunderstood what the Rstudio chart wanted, it is asking for a staging bucket, but I guess that I should feed it the workspace bucket instead of creating a new one. I got confuse with how things are set up in GCE

Reply:
Yeah, I do not see any mention of any of the welder extraEnv that Yonghao added in the chart in leonardo. Looks like Werlder ocnfigs are:
```welder {
  # Set to deploy welder to clusters with the given label
  deployWelderLabel = "saturnVersion"

  # Set to upgrade welder on clusters with the given label
  updateWelderLabel = "saturnVersion"

  # Leo will only deploy welder to clusters created after this date.
  # Clusters created prior to this date will not have welder deployed and will have delocalization disabled.
  deployWelderCutoffDate = "2019-08-01"

  welderReservedMemory = 768m
}```

But that would make any of the Welder-specific values in the chart (google project, staging bucket, cluster name and owner email) obsolete?

Reply:
This is the welder conf, you can see the support env vars:

https://github.com/DataBiosphere/welder/blob/master/server/src/main/resources/reference.conf

Well sure enough, I see
```
staging-bucket-name = ${STAGING_BUCKET}
```

I wonder what it's used for… but anyway this probably answers my question about whether it's actually needed. :)

Reply:
Just talking with Justin, it is used to dump the welder logs!